app/models/cookbook.rb

```diff
@@ -407,4 +401,23 @@ def deprecate_search(query)
   def copy_name_to_lowercase_name
     self.lowercase_name = name.to_s.downcase
   end
+
+  def create_new_collaborator(initiator)
+    unless collaborators.where(user_id: initiator.id).present?
+      collaborator = Collaborator.new(user_id: initiator.id, resourceable: self)
+      collaborator.save!
+    end
```

Comment:
Removed my previous comment after reading deeper in the spec and seeing that we do intend to only create a `Collaborator` for the `initiator` [if that user does not already have collaborator access via a group membership](https://github.com/nellshamrell/supermarket/blob/groups-feature-2/spec/models/cookbook_spec.rb#L209-L227).

I'll did some tinkering and had tests pass with `collaborators.where(user_id: initiator.id).first_or_create!`.